java/serve/src/main/java/io/ray/serve/replica/RayServeReplicaImpl.java

```diff
@@ -218,7 +236,34 @@ private Object[] parseRequestItem(Query requestItem) {
 
     // From other language Proxy or Handle.
     RequestWrapper requestWrapper = (RequestWrapper) requestItem.getArgs();
-    if (requestWrapper.getBody() == null || requestWrapper.getBody().isEmpty()) {
+    // Compatible with some http calls without http ingress
+    if (StringUtils.equals(
+        requestItem.getMetadata().getCallMethodSignature(), Constants.HTTP_PROXY_SIGNATURE)) {
+      LOGGER.info(
+          "body is {}, scope is {}",
+          requestWrapper.getBody().toStringUtf8(),
+          requestWrapper.getScopeMap().toString());
+      if (signatures.containsKey(Constants.HTTP_PROXY_SIGNATURE)) {
+        return new Object[] {requestWrapper};
+      } else if (StringUtils.isNotBlank(requestWrapper.getScopeMap().get("query_string"))) {
+        String[] fields = requestWrapper.getScopeMap().get("query_string").split("&");
+        List<String> args = Lists.newArrayList();
+        for (String field : fields) {
+          String[] tmp = field.split("=");
+          if (tmp.length > 1) {
+            args.add(tmp[1]);
+          }
+        }
+        return args.toArray();
+      } else if (!requestWrapper.getBody().isEmpty()
+          && StringUtils.contains(
+              requestWrapper.getScopeMap().get("headers"), "application/json")) {
+        Gson gson = new Gson();
```

Comment:
Create a reusable gson util, and set the 'disableHtmlEscaping' propertry.

Reply:
Resloved. The gson util class and related unit tests have been added.